### ISSUE TITLE
Need to create local build of CloudCopy to fix build server issue

### ISSUE THREAD

**Issue by Aaronontheweb** [akkadotnet/akka.net#811]
IT'S ALWAYS SOMETHING :(

[CloudCopy](http://cloudcopy.codeplex.com/), the tool we use to uploading Azure hasn't been updated since 2011. And apparently it's not very build-server friendly.

[20:24:06][Step 1/1] Unhandled Exception: System.InvalidOperationException: Cannot see if a key has been pressed when either application does not have a console or when console input has been redirected from a file. Try Console.In.Peek.
[20:24:06][Step 1/1]    at System.Console.get_KeyAvailable()
[20:24:06][Step 1/1]    at CloudCopy.Commands.UploadCommand.Execute()
[20:24:06][Step 1/1]    at CloudCopy.Program.RunCommands()
[20:24:06][Step 1/1]    at CloudCopy.Program.Main(String[] args)

Basically it's trying to do some standard I/O redirection stuff and failing: http://stackoverflow.com/questions/3961542/checking-standard-input-in-c-sharp

I think we can just wrap this in a try...catch block and resolve the issue that way. Either way we'll need to fork the source and hotfix it so our build server can properly upload its API docs to Azure.

**Comment by SeanKilleen** [comment]
@Aaronontheweb would the group be open to documentation? I looked at the docs and since they're already markdown formatted, it might make sense to move them to Github Pages / Jekyll. 

I did this with [my blog](http://seankilleen.com/) [(repo)](https://github.com/SeanKilleen/seankilleen.github.io) -- it builds on the fly to static pages every checkin, all hosted on Github so I don't pay a dime. Also, it's Github's server, so as Zach Holman puts it, "when my site goes down, Github gets paged."

I know it doesn't solve this specific issue, but is there interest in me branching the site and exploring options around that? Seems like it could really reduce some overhead.

**Comment by Aaronontheweb** [comment]
@SeanKilleen the Wiki, I.E. the documentation written by humans, is done on Github pages through drone.io - so we're covered there.

This is the other half of the documentation - the API docs, generated via reflection from all of the XML-DOC comments on the code. It's synthesized automatically by SandCastle.

We considered integrating this stuff using Github pages, but we'd quickly exceed the maximum size of a Github repository (4 gigs I think) just doing our weekly unstable doc builds.

So we decided to go with Azure Blob Storage + CDN for doing this. Simpler to do (easier to do lots of small uploads than one mega commit + push) and just as reliable.

Unfortunately, the command line tool that I've used for doing this in the past (distributed auto updating installers for desktop software this way- it was pretty cool) doesn't play nice with some of the standard I/O redirection that our build server does.

I think it might literally be as simple as wrapping something in a try catch block and suppressing the unhandled exception. That's an easy fix, but just a pain in the butt (CodePlex? COME ON!)

**Comment by SeanKilleen** [comment]
@Aaronontheweb I see, makes sense. Have you thought about installing the azure-cli? You could throw it on the build server, add your publish certain, and then maybe use the azure-cli commands via powershell to push the files...

**Comment by Aaronontheweb** [comment]
@SeanKilleen that's a good idea - think you could modify our build script to do that? If you can, I'll update the build agent image to include azure-cli.

**Comment by SeanKilleen** [comment]
@Aaronontheweb Let me look into how things are structured a little bit first -- don't want to send you on a goose chase, though it should be doable. I'll create a separate issue to track that so we don't lose this one.
